File: js/packages/node/scripts/smoke.mjs
// Real-api smoke: IFOREVENTS_PROJECT_KEY and IFOREVENTS_BASE_URL must be set.
import { createIforevents } from "../dist/index.js";
const projectKey = process.env.IFOREVENTS_PROJECT_KEY;
const baseUrl = process.env.IFOREVENTS_BASE_URL ?? "https://api.iforevents.com";
if (!projectKey) { console.error("IFOREVENTS_PROJECT_KEY missing"); process.exit(2); }
const errors = [];
const { iforevents, api, shutdown } = await createIforevents({ projectKey, baseUrl, batchSize: 2, onError: (e) => errors.push(String(e)) });
const [id] = await iforevents.identify(`smoke_node_${Date.now()}`, { email: "smoke@example.com", plan: "free" });
await iforevents.track("smoke_track", { n: 1 });
await iforevents.page("/smoke");
await shutdown();
console.log(JSON.stringify({ identify: id.success, userId: api.currentUserId, queued: api.queuedEventsCount, errors }));
process.exit(id.success && errors.length === 0 ? 0 : 1);

File: java/iforevents/src/test/java/com/iforevents/Smoke.java
package com.iforevents;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/** Real-api smoke: IFOREVENTS_PROJECT_KEY and IFOREVENTS_BASE_URL must be set. */
public final class Smoke {
    public static void main(String[] args) throws Exception {
        String key = System.getenv("IFOREVENTS_PROJECT_KEY");
        String base = System.getenv("IFOREVENTS_BASE_URL");
        if (base == null || base.isEmpty()) base = "https://api.iforevents.com";
        if (key == null || key.isEmpty()) {
            System.err.println("IFOREVENTS_PROJECT_KEY missing");
            System.exit(2);
        }
        final List<String> errors = new ArrayList<String>();
        IForeventsAPIIntegration api = new IForeventsAPIIntegration(APIConfig.builder(key).baseUrl(base).batchSize(2).flushOnShutdownHook(false).onError(new APIConfig.Callback<IForeventsAPIException>() {
            @Override
            public void call(IForeventsAPIException e) {
                errors.add(e.getMessage());
            }
        }).build());
        Iforevents ife = Iforevents.builder().integration(api).build();
        ife.init();
        Map<String, Object> traits = new LinkedHashMap<String, Object>();
        traits.put("email", "smoke@example.com");
        traits.put("plan", "free");
        Map<String, Object> nested = new LinkedHashMap<String, Object>();
        nested.put("deep", true);
        traits.put("nested", nested);
        List<IntegrationResult> id = ife.identify("smoke_java_" + System.currentTimeMillis(), traits);
        Map<String, Object> props = new LinkedHashMap<String, Object>();
        props.put("n", 1);
        ife.track("smoke_track", props);
        ife.page("/smoke", null);
        ife.shutdown();
        System.out.println("{\"identify\":" + id.get(0).success + ",\"user_id\":\"" + api.userId() + "\",\"queued\":" + api.queuedEvents() + ",\"errors\":" + errors + "}");
        System.exit(id.get(0).success && errors.isEmpty() && api.userId() != null ? 0 : 1);
    }
}
